rest_framework/serializers.py

```diff
@@ -511,7 +511,10 @@ def to_representation(self, instance):
             if check_for_none is None:
                 ret[field.field_name] = None
             else:
-                ret[field.field_name] = field.to_representation(attribute)
+                try:
+                    ret[field.field_name] = field.to_representation(attribute)
+                except SkipField:
+                    continue
```

Comment:
hi, @davidmgvaz do we have some test to cover this exception?

Reply:
This exception should be raised whenever you want to skip a field for whatever reason e.g: 

```
class ModelSerializer(serializers.ModelSerializer):
    detail = serializers.SerializerMethodField(required=False)

    class Meta:
        model = Model
        fields = (
            ...
            'detail'
        )

    def get_detail(self, obj):
        if not obj.needs_detail:
            raise SkipField('Not Needed')
        ...
        return detail
```